pallets/settlement/src/lib.rs

```diff
@@ -664,7 +553,7 @@ decl_module! {
         ///
         /// # Weight
         /// `950_000_000 + 1_000_000 * legs.len()`
-        #[weight = weight_for::weight_for_instruction_creation::<T>(legs.len())]
+        #[weight = <T as Trait>::WeightInfo::add_instruction_with_settle_on_block_type(legs.len() as u32)]
```

Comment:
This should include the execution weight.

Reply:
Yeah, I leave it on purpose. I will add this in another PR once we are done with all other hassles.